### PR TITLE
Add `selected_resources` Jinja context to docs - v1.1

### LINKED ISSUE #1330: Add documentation about `selected_resources` (releasing with 1.1)
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/dbt-jinja-functions
a new page also needs to be created to provide description for the variable

### What part(s) of the page would you like to see updated?

`selected_resources` will be a variable available in Jinja in dbt 1.1 (via PR [3471](https://github.com/dbt-labs/dbt-core/issues/3471)) that will provide a list of all selected resources based on the selectors used

Updates required are:
- the page with all [dbt Jinja functions](https://docs.getdbt.com/reference/dbt-jinja-functions) needs to be updated to include `selected_resources` 
- **and** a new page needs to be created to provide information on what the variable contains, with similar caveats as `graph` on when the variable contains information or not

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by b-per**
closes #1330 
## Description & motivation

PR for the issue #1330 , adding a variable called `selected_resources` to the Jinja context

## To-do before merge

- [x] Change the base branch
- [x] Ensure PR [#5001](https://github.com/dbt-labs/dbt-core/pull/5001) is merged in dbt-core

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

**Comment by runleonarun** [comment]
@b-per Thank you for opening this PR! I changed the branch to `current` because now that we can version the docs, we've been releasing 1.1 content into `current` and versioning it like we do [here](https://github.com/dbt-labs/docs.getdbt.com/pull/1319/files#diff-52822271cc0a418238ec087115df4ef8b81bf2c5d5b2eb520efad49e1c62cb95R106). I can jump in an add the versioning!

File: website/docs/reference/dbt-jinja-functions/selected_resources.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Currently, this variable is not accessible when using the command `run-operation`.
+
+:::danger Heads up
+
+dbt actively builds the graph during the [parsing phase](execute) of
+running dbt projects, so the `selected_resources` context variable will be
+empty during parsing. Please read the information below carefully
+to understand how to effectively use this variable.
+
+:::
```
Comment (jtcohen6) [inline website/docs/reference/dbt-jinja-functions/selected_resources.md]: 🙏 thank you

File: website/sidebars.js [hunk 2]
```diff
@@ -510,51 +510,16 @@ const sidebarSettings = {
...
-            "reference/dbt-jinja-functions/target",
-            "reference/dbt-jinja-functions/this",
-            "reference/dbt-jinja-functions/tojson",
-            "reference/dbt-jinja-functions/toyaml",
-            "reference/dbt-jinja-functions/var",
-            "reference/dbt-jinja-functions/profiles-yml-context",
-            "reference/dbt-jinja-functions/dbt-project-yml-context",
-            "reference/dbt-jinja-functions/on-run-end-context",
-          ],
+          type: "autogenerated",
```
Comment (jtcohen6) [inline website/sidebars.js:523]: whoa - very cool

File: website/dbt-versions.js [hunk 3]
```diff
@@ -26,5 +26,9 @@ exports.versionedPages = [
   {
     "page": "docs/guides/migration-guide/upgrading-to-v1.1",
     "firstVersion": "1.1",
+  },
+  {
+    "page": "reference/dbt-jinja-functions/print/selected_resources",
```
Comment (runleonarun) [inline website/dbt-versions.js]:
```suggestion
    "page": "reference/dbt-jinja-functions/selected_resources",
```

File: website/dbt-versions.js [hunk 4]
```diff
@@ -27,8 +27,12 @@ exports.versionedPages = [
...
   },
+  {
+    "page": "reference/dbt-jinja-functions/selected_resources",
+    "firstVersion": "1.1",
+  },
   {
     "page": "reference/dbt-jinja-functions/print",
     "firstVersion": "1.1",
-  }
+  },
```
Comment (runleonarun) [inline website/dbt-versions.js]:
```suggestion
  }
```

File: website/docs/reference/dbt-jinja-functions/selected_resources.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+title: "selected_resources"
+id: "selected_resources"
+---
+
+The `selected_resources` context variable contains a list of all the _nodes_ 
+selected by the current dbt command. 
+
+Currently, this variable is not accessible when using the command `run-operation`.
+
+:::danger Heads up
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/selected_resources.md]:
```suggestion
:::danger Warning!
```

File: website/docs/reference/dbt-jinja-functions/selected_resources.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+dbt actively builds the graph during the [parsing phase](execute) of
+running dbt projects, so the `selected_resources` context variable will be
+empty during parsing. Please read the information below carefully
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/selected_resources.md]:
```suggestion
empty during parsing. Please read the information on this page to effectively use this variable.
```

File: website/docs/reference/dbt-jinja-functions/selected_resources.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+to understand how to effectively use this variable.
```
Comment (runleonarun) [inline website/docs/reference/dbt-jinja-functions/selected_resources.md]:
```suggestion
```

**Review by runleonarun — APPROVED** [review]
Just a few tiny edits

**Pull request merged**